include/alpaka/mem/buf/sycl/Copy.hpp

```diff
@@ -21,129 +21,367 @@
+            TExtent const& extent)
+            -> alpaka::detail::TaskCopySycl<TDim, std::remove_reference_t<TViewDstFwd>, TViewSrc, TExtent>
         {
             ALPAKA_DEBUG_FULL_LOG_SCOPE;
 
-            constexpr auto copy_dim = static_cast<int>(Dim<TExtent>::value);
-            using ElemType = Elem<std::remove_const_t<TViewSrc>>;
-            using SrcType = ElemType const*;
-            using DstType = alpaka::experimental::detail::DstAccessor<ElemType, copy_dim>;
+            return {std::forward<TViewDstFwd>(viewDst), viewSrc, extent};
+        }
+    };
 
-            auto const range = experimental::detail::make_sycl_range(ext);
-            auto const offset = experimental::detail::make_sycl_offset(viewDst);
+    //! The SYCL device-to-host memory copy trait specialization.
+    template<typename TPltf, typename TDim>
+    struct CreateTaskMemcpy<TDim, DevCpu, DevGenericSycl<TPltf>>
+    {
+        template<typename TExtent, typename TViewSrc, typename TViewDstFwd>
+        ALPAKA_FN_HOST static auto createTaskMemcpy(
+            TViewDstFwd&& viewDst,
+            TViewSrc const& viewSrc,
+            TExtent const& extent)
+            -> alpaka::detail::TaskCopySycl<TDim, std::remove_reference_t<TViewDstFwd>, TViewSrc, TExtent>
+        {
+            ALPAKA_DEBUG_FULL_LOG_SCOPE;
 
-            return experimental::detail::TaskCopySycl<SrcType, DstType, experimental::detail::Direction::h2d>{
-                getPtrNative(viewSrc),
-                DstType{viewDst.m_buffer, range, offset}};
+            return {std::forward<TViewDstFwd>(viewDst), viewSrc, extent};
         }
     };
 
-    //! The SYCL device-to-host memory copy trait specialization.
-    template<typename TDim, typename TPltf>
-    struct CreateTaskMemcpy<TDim, DevCpu, experimental::DevGenericSycl<TPltf>>
+    //! The SYCL device-to-device memory copy trait specialization.
+    template<typename TPltfDst, typename TPltfSrc, typename TDim>
+    struct CreateTaskMemcpy<TDim, DevGenericSycl<TPltfDst>, DevGenericSycl<TPltfSrc>>
     {
         template<typename TExtent, typename TViewSrc, typename TViewDstFwd>
-        static auto createTaskMemcpy(TViewDstFwd&& viewDst, TViewSrc const& viewSrc, TExtent const& ext)
+        ALPAKA_FN_HOST static auto createTaskMemcpy(
+            TViewDstFwd&& viewDst,
+            TViewSrc const& viewSrc,
+            TExtent const& extent)
+            -> alpaka::detail::TaskCopySycl<TDim, std::remove_reference_t<TViewDstFwd>, TViewSrc, TExtent>
+        {
+            ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+            return {std::forward<TViewDstFwd>(viewDst), viewSrc, extent};
+        }
+    };
+
+    //! The SYCL non-blocking device queue scalar copy enqueue trait specialization.
+    template<typename TPltf, typename TExtent, typename TViewSrc, typename TViewDst>
+    struct Enqueue<
+        alpaka::QueueGenericSyclNonBlocking<TPltf>,
+        alpaka::detail::TaskCopySycl<DimInt<0u>, TViewDst, TViewSrc, TExtent>>
+    {
```

Comment:
Why are you adding specializations for the `Enqueue` trait? Shouldn't the existing design (using objects that can be used as SYCL command groups) also fit for this use case?

Reply:
I don't think the existing design without the specialization would work because we are using the `memcpy` method of the `sycl::queue`, not of the `sycl::handler` as it was before.

Reply:
And is there a reason for not calling `cgh.memcpy` and using the queue version instead?

Reply:
In both cases, moving from the `sycl::queue`'s methods to the `sycl::handler`'s methods we lose the ability to Copy/Set in N dimensions as we are doing right now. We would need a specialised method for that since one handler task can call a single memory operation / kernel